examples/multigpu_advtrain/resnet_tf.py

```diff
@@ -0,0 +1,354 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import namedtuple
+
+import six
+
+from cleverhans.model import Model
+from collections import OrderedDict
+
+import tensorflow as tf
+
+from model import ReLU
+
+HParams = namedtuple('HParams',
+                     'batch_size, num_classes, min_lrn_rate, lrn_rate, '
+                     'num_residual_units, use_bottleneck, weight_decay_rate, '
+                     'relu_leakiness, optimizer, momentum')
+
+
+class ResNetTF(Model):
+    """ResNet model."""
+
+    def __init__(self, batch_size=None, name=None, optimizer='mom',
+                 *args, **kwargs):
+        self.global_step = tf.contrib.framework.get_or_create_global_step()
+        self.hps = HParams(batch_size=batch_size,
+                           num_classes=10,
+                           min_lrn_rate=0.0001,
+                           lrn_rate=0.1,
+                           num_residual_units=5,
+                           # num_residual_units=4,
+                           use_bottleneck=False,
+                           weight_decay_rate=0.0002,
+                           relu_leakiness=0.1,
+                           optimizer=optimizer,
+                           momentum=.9)
+        self.reuse = None
+        self.kwargs = {}
+        self.init_kwargs = kwargs
+        del self.init_kwargs['input_shape']
+        self.layers = []
+        self.layer_idx = 0
+        self.init_layers = True
+        self.decay_cost = None
+        self.training = None
+        self.device_name = None
+
+        self._extra_train_ops = []
+
+    def get_layer_names(self):
+        return ['logits', 'probs']
+
+    def set_device(self, device_name):
+        self.device_name = device_name
+        for layer in self.layers:
+            layer.device_name = device_name
+
+    def fprop(self, x, return_all=False, dataset='cifar10', **kwargs):
+        if 'training' in kwargs:
+            self.training = kwargs['training']
+        self.kwargs = kwargs
+        self.kwargs.update(self.init_kwargs)
+        if 'input_shape' in self.kwargs:
+            del self.kwargs['input_shape']
+        self.layer_idx = 0
+        with tf.variable_scope('Resnet', reuse=self.reuse):
+            logits, probs = self._build_model(x)
+        if self.Conv2D is None:
+            self.reuse = True
+        else:
+            # else my own conv is used and no reuse is needed
```

Comment:
Reword this comment, it won't be clear who "my" refers to in the future

Reply:
Fixed in d1169ec.